File: artexploration/src/main/java/com/example/artexploration/part3/TestActivity.java
package com.example.artexploration.part3;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import com.example.artexploration.R;

public class TestActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test2);
        MyView button = findViewById(R.id.bt_test);
        button.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                Log.d("ccg", "onTouch:"+event.getAction());
                return false;
            }
        });
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.d("ccg", "onClick");
            }
        });
    }
}

File: developers/src/main/java/com/example/developers/jetpack/livedateAviewmodel/TestActivity.kt
package com.example.developers.jetpack.livedateAviewmodel

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.example.developers.R

class TestActivity : AppCompatActivity() {

    private lateinit var tvNum:TextView
    private lateinit var btAdd:Button
    private lateinit var model: TestViewModel


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_test2)
        model = ViewModelProviders.of(this).get(TestViewModel::class.java)
        val numObserver = Observer<Int> { newName ->
            tvNum.text = newName.toString()
        }
        model.currentName?.observe(this, numObserver)
        initView()
    }

    private fun initView() {
        tvNum = findViewById(R.id.tvNum)
        btAdd = findViewById(R.id.btAdd)
        btAdd.setOnClickListener {
            model.addNum(1)
        }

    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("ccg","destroy")
    }
}
